packages/zoe/src/contracts/simpleExchange.js

```diff
@@ -77,7 +77,15 @@ export const makeContract = harden(
     // the caller can know to add the new offer to the book.
     function swapIfCanTrade(offerHandles, offerHandle) {
       for (const iHandle of offerHandles) {
-        if (canTradeWith(offerHandle, iHandle)) {
+        const iHandleSatisfied = satisfies(
+          iHandle,
+          zcf.getCurrentAllocation(offerHandle),
+        );
+        const offerHandleSatisfied = satisfies(
+          offerHandle,
+          zcf.getCurrentAllocation(iHandle),
+        );
+        if (iHandleSatisfied && offerHandleSatisfied) {
```

Comment:
Seems like pulling out the common code would simplify this. But maybe after eslint gets through with it it wouldn't be any shorter or easier to read.  Just an idea.

```
  function handleSatisfiedBy(xHandle, yHandle) {
    return satisfies(xHandle, zcf.getCurrentAllocation(yHandle));
  }
 if (satisfiedBy(iHandle, offerHandle) && satisfiedBy(offerHandle, iHandle)) {
```

Reply:
Done